components/hybrid_engine/src/engine.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2023 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_traits::{KvEngine, MemoryEngine, Peekable, ReadOptions, Result, SyncMutable};
+
+use crate::snapshot::HybridSnapshot;
+
+/// This engine is structured with both a disk engine and an in-memory engine.
+/// The disk engine houses the complete database data, whereas the in-memory
+/// engine functions as a region cache, selectively caching certain regions to
```

Comment:
On engine trait level, it doesn't have knowledge about region, how does it know which ranges should be cached.